samplers/aws/xray/remote_sampler.go

```diff
@@ -209,14 +201,15 @@ func (rs *RemoteSampler) refreshManifest(ctx context.Context) (err error) {
 		err = errors.New("error occurred creating/updating rules")
 	}
 
-	// Prune inactive rules
-	rs.manifest.prune(actives)
+	// Re-sort to fix matching priorities.
+	tempManifest.sort()
 
-	// Re-sort to fix matching priorities
-	rs.manifest.sort()
+	// assign temp manifest to original copy/one sync refresh.
+	rs.manifest.mu.Lock()
+	rs.manifest.rules = tempManifest.rules
```

Comment:
This should look something like `rs.manifest = tempManifest` rather than assigning to multiple fields, probably by moving the lock up to the sampler instead of the manifest.